packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,295 @@
+import { WALLET_CONFIG } from './wallet.tokens';
+import { WalletConfig, WalletState } from './wallet.types';
+
+export const WALLET_DEFAULT_CONFIG: WalletConfig = {
+    wallets: [],
+    autoConnect: false,
+    localStorageKey: 'walletName',
+    onError: (error: unknown) => console.error(error),
+};
+
+@Injectable()
+export class WalletStore extends ComponentStore<WalletState> {
+    private readonly _autoConnect = this._config?.autoConnect || false;
+    private readonly _localStorageKey = this._config?.localStorageKey || 'walletName';
+    private logError = this._config?.onError || ((error: unknown) => console.error(error));
+    readonly wallets$ = this.select((state) => state.wallets);
+    readonly selectedWallet$ = this.select((state) => state.selectedWallet);
+    readonly connected$ = this.select((state) => state.connected);
+    readonly connecting$ = this.select((state) => state.connecting);
+    readonly disconnecting$ = this.select((state) => state.disconnecting);
+    readonly wallet$ = this.select((state) => state.wallet);
+    readonly adapter$ = this.select((state) => state.adapter);
+    readonly publicKey$ = this.select((state) => state.publicKey);
+    readonly ready$ = this.select((state) => state.ready);
+    readonly anchorWallet$ = this.select(
+        this.publicKey$.pipe(isNotNull),
+        this.adapter$.pipe(isNotNull),
+        (publicKey, adapter) => {
+            return publicKey && 'signTransaction' in adapter && 'signAllTransactions' in adapter
+                ? {
+                      publicKey,
+                      signTransaction: (transaction: Transaction) => this.signTransaction(transaction)?.toPromise(),
+                      signAllTransactions: (transactions: Transaction[]) =>
+                          this.signAllTransactions(transactions)?.toPromise(),
+                  }
+                : undefined;
+        }
+    );
+
+    constructor(
+        @Optional()
+        @Inject(WALLET_CONFIG)
+        private _config: WalletConfig
+    ) {
+        super({
+            wallets: _config.wallets,
+            selectedWallet: null,
+            wallet: null,
+            adapter: null,
+            connected: false,
+            connecting: false,
+            disconnecting: false,
+            ready: false,
+            publicKey: null,
+            autoApprove: false,
+        });
+
+        if (!this._config) {
+            this._config = WALLET_DEFAULT_CONFIG;
+        } else {
+            this._config = {
+                ...WALLET_DEFAULT_CONFIG,
+                ...this._config,
+            };
+        }
+
+        const walletName = localStorage.getItem(this._localStorageKey);
+        const wallet = this._config.wallets.find(({ name }) => name === walletName);
+
+        if (wallet) {
+            this.selectWallet(walletName as WalletName);
+        } else if (this._config.wallets.length > 0) {
+            this.selectWallet(this._config.wallets[0].name);
```

Comment:
The reason behind this is that I wanted to **always** have a selected wallet, thinking it through, this is probably something we want consumers to choose instead of imposing it.

Reply:
Yeah, what if the first wallet on the list doesn't work on the user's browser/device, or they just don't use it? Also, a UI would then show that wallet as selected when they didn't do anything.